netty/src/test/java/io/grpc/netty/ProtocolNegotiatorsTest.java

```diff
@@ -0,0 +1,213 @@
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.netty;
+
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.Iterables;
+
+import io.grpc.netty.ProtocolNegotiators.TlsChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelPipeline;
+import io.netty.handler.ssl.SslHandler;
+import io.netty.handler.ssl.SslHandshakeCompletionEvent;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import java.util.logging.Filter;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+
+@RunWith(JUnit4.class)
+public class ProtocolNegotiatorsTest {
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+  @Rule public final MockitoRule mocks = MockitoJUnit.rule();
+
+  @Mock private ChannelHandlerContext channelHandlerCtx;
+  @Mock private ChannelPipeline pipeline;
+  @Mock private ChannelHandler channelHandler;
+
+  private SSLEngine engine;
+
+  @Before
+  public void setUp() throws Exception {
+    engine = SSLContext.getDefault().createSSLEngine();
+    engine.getSession();
+  }
+
+  @Test
+  public void tlsHandler() throws Exception {
```

Comment:
I don't want to "give a false sense of coverage".

Consider having 100% coverage but learning that there was not a single assert. That would be a false sense of coverage. It would be "already written, not wrong, and up the coverage count," but also likely not fulfilling its primary purpose to make sure that the code being tested is correct. Having a fake test is worse in my mind than not having a test at all.

Also, it [does seem it is already covered](https://coveralls.io/builds/3843611/source?filename=netty%2Fsrc%2Fmain%2Fjava%2Fio%2Fgrpc%2Fnetty%2FProtocolNegotiators.java#L84), I believe due to tests in interop-testing (which I admit is not a true unit test). I would rather notice the uncovered line if I ever ran ProtocolNegotiatorsTest by itself.